Code/GraphMol/SynthonSpaceSearch/SynthonSet.cpp

```diff
@@ -382,6 +451,69 @@ void SynthonSet::buildSynthonFingerprints(
   }
 }
 
+void SynthonSet::buildAddAndSubtractFPs(
+    const FingerprintGenerator<std::uint64_t> &fpGen,
+    const SynthonSpaceSearchParams &params) {
+  d_addFP.reset();
+  d_subtractFP.reset();
+  auto lastAddFP = std::make_unique<ExplicitBitVect>();
+  auto lastsubtractFP = std::make_unique<ExplicitBitVect>();
+  std::unique_ptr<boost::mt19937> randGen;
+  if (params.randomSeed == -1) {
+    std::random_device rd;
+    randGen = std::make_unique<boost::mt19937>(rd());
+  } else {
+    randGen = std::make_unique<boost::mt19937>(params.randomSeed);
+  }
+  std::vector<boost::random::uniform_int_distribution<size_t>> synthSels;
```

Comment:
do you really need a vector of distributions? Can't you just have one?

Reply:
There are different sized lists of synthons.